projects/berbix/src/lib/berbix.component.ts

```diff
@@ -1,153 +1,195 @@
+      flowStateChange,
+      flowExit,
+    } = this;
 
     if (e.origin !== this.getBaseUrl()) {
       return;
     }
 
     const data = JSON.parse(e.data);
-    if (data.type === 'VERIFICATION_COMPLETE') {
+    if (data.type === "VERIFICATION_COMPLETE") {
       try {
         if (data.payload.success) {
-          flowCompleted.emit({ code: data.payload.code });
+          flowCompleted.emit(null);
         } else {
           flowError.emit(data);
         }
       } catch (e) {
         // Continue clean-up even if callback throws
       }
       this.show = false;
-    } else if (data.type === 'DISPLAY_IFRAME') {
+    } else if (data.type === "DISPLAY_IFRAME") {
       flowDisplayed.emit(null);
       this.height = data.payload.height;
-    } else if (data.type === 'RESIZE_IFRAME') {
+      this.marginTop = data.payload.margin || 0;
+    } else if (data.type === "RESIZE_IFRAME") {
       this.height = data.payload.height;
-    } else if (data.type === 'RELOAD_IFRAME') {
+    } else if (data.type === "RELOAD_IFRAME") {
       this.height = 0;
       this.idx += 1;
-    } else if (data.type === 'STATE_CHANGE') {
+    } else if (data.type === "STATE_CHANGE") {
       flowStateChange.emit(data.payload);
-    } else if (data.type === 'ERROR_RENDERED') {
+    } else if (data.type === "EXIT_MODAL") {
+      flowExit.emit(null);
+    } else if (data.type === "ERROR_RENDERED") {
       flowError.emit(data.payload);
       this.height = 200;
     }
-  }
+  };
 
   getBaseUrl() {
-    const { baseUrl, environment } = this;
+    const { baseUrl } = this;
     if (baseUrl != null) {
       return baseUrl;
     }
-    switch (environment) {
-      case 'sandbox':
-        return 'https://verify.sandbox.berbix.com';
-      case 'staging':
-        return 'https://verify.staging.berbix.com';
-      default:
-        return 'https://verify.berbix.com';
-    }
+    return "https://verify.berbix.com";
   }
 
   getFrameUrl() {
-    const { overrideUrl, version, clientId, role, templateKey, email, phone, continuation, clientToken } = this;
+    const {
+      overrideUrl,
+      version,
+      clientToken,
+      showInModal,
+      showCloseModalButton,
+    } = this;
     if (overrideUrl != null) {
       return overrideUrl;
     }
-    const token = clientToken || continuation;
-    const template = templateKey || role;
-    var options = ['sdk=BerbixAngular-' + SDK_VERSION];
-    if (clientId) {
-      options.push('client_id=' + clientId);
-    }
-    if (template) {
-      options.push('template=' + template);
-    }
-    if (email) {
-      options.push('email=' + encodeURIComponent(email));
-    }
-    if (phone) {
-      options.push('phone=' + encodeURIComponent(phone));
-    }
+    const token = clientToken;
+    var options = ["sdk=BerbixAngular-" + SDK_VERSION];
     if (token) {
-      options.push('client_token=' + token);
+      options.push("client_token=" + token);
     }
+    if (showInModal) {
+      if (!showCloseModalButton) {
```

Comment:
For readability, it's preferred to use the affirmative in the main condition of an if-else, i.e. `if(showCloseModalButton)`.